.github/workflows/code-coverage.yml

```diff
@@ -13,18 +24,19 @@ jobs:
         run: yarn install
       - uses: actions/setup-go@v2
         with:
-          go-version: '1.17.8'
+          go-version: "1.17.8"
       - name: generate go for github.com/grafana/grafana
         if: github.repository == 'grafana/grafana'
         run: make gen-go
 
       - name: Get go coverage for caller's PR
+        id: pr-backend
         run: |
-          go test -v -coverpkg=./... -coverprofile=profile.cov ./...
-          echo "go-pr-cov=$(go tool cover -func profile.cov | grep total: | awk '{print substr($NF,1,length($NF)-1)}')" >> $GITHUB_ENV
+          echo "::set-output name=coverage::$(./scripts/ci-code-coverage.sh false "${{steps.relevant-paths.outputs.backend-paths}}")"
```

Comment:
the problem of using an external script is that:

- You are referring an external file that you have no control from in this file
- You need to copy the script for every repo that wants to use this (with potential differences)

I would just dump the script code here, it's more difficult to read but you can control what gets executed without repetition.

Reply:
This makes complete sense, I wasn't particularly happy putting the scripts in the Grafana repo. Will switch this round!